webpack/webpack5-module-minifier-plugin/src/AsyncImportCompressionPlugin.ts

```diff
@@ -0,0 +1,314 @@
+                new WebpackError(`Unexpected async import ${token} in module ${context.loggingName}`)
+              );
+              return source;
+            }
+          }
+
+          const localImport: ILocalImportMetadata | undefined = localImports.get(token);
+          if (!localImport) {
+            context.compilation.errors.push(
+              new WebpackError(`Missing metadata for ${token} in module ${context.loggingName}`)
+            );
+            return source;
+          }
+          const { meta, module } = localImport;
+
+          const chunkExpression: string = meta.index < 0 ? JSON.stringify(meta.chunkIds) : `${meta.index}`;
+
+          source.replace(
+            match.index,
+            ASYNC_IMPORT_REGEX.lastIndex - 1,
+            `${chunkExpression},${JSON.stringify(module.id!)}${getImportTypeExpression(
+              module,
+              context.module!,
+              moduleGraph
+            )}`
+          );
+        }
+
+        return source;
+      });
+
+      compilation.hooks.beforeChunkAssets.tap({ name: PLUGIN_NAME, stage: STAGE_AFTER }, () => {
+        const ImportDependency: typeof WebpackImportDependency = getImportDependency(compilation);
+
+        for (const module of compilation.modules) {
+          let toProcess: Module[];
+
+          if (isConcatenatedModule(module)) {
+            toProcess = module.modules;
+          } else {
+            toProcess = [module];
+          }
+
+          for (const child of toProcess) {
+            processModuleDependenciesRecursive(child, (dep) => {
+              if (dep instanceof ImportDependency) {
+                const targetModule: Module = moduleGraph.getModule(dep);
+
+                if (targetModule) {
+                  let localAsyncImports: Map<string, ILocalImportMetadata> | undefined =
+                    asyncImportMap.get(module);
+                  if (!localAsyncImports) {
+                    asyncImportMap.set(module, (localAsyncImports = new Map()));
+                  }
+
+                  const chunkGroups: ChunkGroup[] = targetModule.blocks.map(chunkGraph.getBlockChunkGroup);
+                  const chunkIds: Set<number | string | null> = new Set();
+
+                  for (const chunkGroup of chunkGroups) {
+                    for (const chunk of chunkGroup.chunks) {
+                      chunkIds.add(chunk.id);
+                    }
+                  }
+
+                  const idString: string = Array.from(chunkIds).join(';');
+
+                  let meta: IAsyncImportMetadata | undefined = asyncImportGroups.get(idString);
+                  if (!meta) {
+                    asyncImportGroups.set(
+                      idString,
+                      (meta = { chunkCount: chunkIds.size, chunkIds: [], count: 0, index: -1 })
+                    );
+                  }
+
+                  meta.count++;
+
+                  const stringKey: string = `${targetModule.id}`.replace(/[^A-Za-z0-9_$]/g, '_');
```

Comment:
`targetModule.id` is deprecated. Value has to come from `chunkGraph.getModuleId(targetModule)`